internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -77,16 +77,27 @@ func ProbeWithCallback(disk string, options options.ProbeOptions, callback func(
 
 	// here we need to read the EFI vars to see if we have any defaults
 	// and populate config accordingly
-	// https://www.freedesktop.org/software/systemd/man/systemd-boot.html#LoaderEntryDefault
-	// this should be set on install/upgrades
+	// https://www.freedesktop.org/software/systemd/man/latest/systemd-boot.html#LoaderEntrySelected
+	// this is set by systemd-boot.
 	efiCtx := efivario.NewDefaultContext()
 
+	// first we start by checking if we have a Selected entry
+	// this is set by systemd-boot when we select an entry
 	bootedEntry, err := ReadVariable(efiCtx, LoaderEntrySelectedName)
 	if err != nil {
 		return nil, err
 	}
 
-	config := &Config{}
+	if bootedEntry == "" {
+		// if we don't have a selected entry, we check the default entry
+		// this is set by systemd-boot when we boot
```

Comment:
is it _set_ by us? (or should be?) or whoever installs UKIs?

Reply:
yeh install code sets this